### DISCUSSION TITLE
Is there a mechanism to include elements by reference?

### DISCUSSION THREAD

**Discussion by SybelBlue** [PrairieLearn/PrairieLearn#6305] [Q&A] [answered]
Hello! 

I'm aware that there is a way to make elements available to a single course or across many courses in the same repository, but is there a mechanism that allows for including elements by reference, à la autograders with docker or element dependencies with node?

Several instructors have been writing custom elements that they want to use across forks of prairielearn, and don't want to copy/paste code between instances or use git submodules that isolate the custom element from the development environment.

Cheers,
Logan

**Comment by jonatanschroeder** [comment, marked as answer]
There is currently no way to share resources (including elements) between courses (though see discussion in #5907 and #6144). However, if you have an element that you believe may be useful for the community at large, you're welcome to submit a PR to make it accessible as an official element.

**Comment by mwest1066** [comment]
We are actively working on question sharing and element sharing so in the future we should have a nice built-in way of handling this (but for now @jonatanschroeder's answer is the best we can do).

**Comment by SybelBlue** [comment]
Thank you both!
